tracing/exporter.go

```diff
@@ -0,0 +1,80 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package tracing
+
+import (
+	gocontext "context"
+
+	srvconfig "github.com/containerd/containerd/services/server/config"
+	"github.com/pkg/errors"
+
+	"go.opentelemetry.io/otel/exporters/otlp"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpgrpc"
+	"go.opentelemetry.io/otel/exporters/stdout"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+func initStdoutExporter(ctx gocontext.Context, cfg srvconfig.OpenTelemetryConfig) (sdktrace.SpanExporter, error) {
+	err := ctx.Err()
+
+	if cfg.ExporterName != "stdout" {
+		return nil, errors.Wrapf(err, "failed to initialize stdout exporter")
+	}
+
+	exporter, err := stdout.NewExporter(
+		stdout.WithPrettyPrint(),
+	)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to initilize stdout exporter")
+	}
+
+	return exporter, nil
+}
+
+func initOtlpExporter(ctx gocontext.Context, cfg srvconfig.OpenTelemetryConfig) (sdktrace.SpanExporter, error) {
+	exp, err := otlp.NewExporter(ctx, otlpgrpc.NewDriver(
+		otlpgrpc.WithInsecure(),
+		otlpgrpc.WithEndpoint(cfg.ExporterEndPoint),
+	))
```

Comment:
@Aneurysm9 @rakyll Do you think it is better to support both OLTP/gRPC and OLTP/HTTP?

Reply:
would be nice to know what protocols are being enabled.. what the options are..